.github/workflows/ci.yml

```diff
@@ -0,0 +1,159 @@
+  push:
+    branches: [ 'dblp' ]
+  pull_request:
+    branches: [ 'dblp' ]
+  workflow_dispatch:
+    branches: '*'
+    inputs:
+      tag:
+        description: 'Add new GitHub tag'
+        required: true
+        type: boolean
+
+env:
+  GIT_TOKEN: ${{ secrets.GIT_TOKEN }}
+
+jobs:
+  lint-test-build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Use Node.js 16
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: Install npm dependencies
+        run: |
+          npm run security-checks
+          npm ci
+
+      - name: Lint
+        run: npm run lint
+
+      - name: Build packages dependencies
+        run: npm run build-dependencies
+
+      - name: Test locales
+        run: npm run test-locales
+
+      - name: Test lambda
+        run: npm run test-lambda
+
+      - name: Test sundial
+        run: npm run test-sundial
+
+      - name: Test tideline
+        run: npm run test-tideline
+
+      - name: Test medical domain
+        run: npm run test-medical-domain
+
+      - name: Test dumb
+        run: npm run test-dumb
+
+      - name: Test viz
+        run: npm run test-viz
+
+      - name: Test blip
+        run: npm run test-blip
+
+      - name: Test yourloops unit
+        run: npm run test-yourloops-unit
+
+      - name: Test yourloops integration
+        run: npm run test-yourloops-integration
+
+      - name: Build
+        run: nice bash build.sh
+
+      - name: Upload yourloops build
+        if: github.event_name != 'pull_request'
+        uses: actions/upload-artifact@v3
+        with:
+          name: dist
+          path: dist/
+
+  publish:
+    if: github.event_name != 'pull_request'
+    needs: lint-test-build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Use Node.js 16
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: Install npm dependencies
+        run: npm ci
+
+      - name: Download artifact
+        uses: actions/download-artifact@v3
+
+      - name: Get yourloops version from package.json
+        id: package-version
+        uses: martinbeentjes/npm-get-version-action@main
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to the Nexus registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ secrets.REGISTRY }}
+          username: ${{ secrets.NEXUS_USERNAME }}
+          password: ${{ secrets.NEXUS_PASSWORD }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ secrets.REGISTRY }}/blip
+          tags: type=ref,enable=true,prefix=${{ steps.package-version.outputs.current-version }}-PR-,event=pr
```

Comment:
What if the event is not a PR? This means we can only publish on PR branches?

Reply:
👍🏻